### PR TITLE
Meeting notes 2023-05-29

### LINKED ISSUE #85: Node.js  Performance Team Meeting 2023-05-29
## Time

**UTC Mon 29-May-2023 17:00 (05:00 PM)**:

| Timezone      | Date/Time             |
|---------------|-----------------------|
| US / Pacific  | Mon 29-May-2023 10:00 (10:00 AM) |
| US / Mountain | Mon 29-May-2023 11:00 (11:00 AM) |
| US / Central  | Mon 29-May-2023 12:00 (12:00 PM) |
| US / Eastern  | Mon 29-May-2023 13:00 (01:00 PM) |
| EU / Western  | Mon 29-May-2023 18:00 (06:00 PM) |
| EU / Central  | Mon 29-May-2023 19:00 (07:00 PM) |
| EU / Eastern  | Mon 29-May-2023 20:00 (08:00 PM) |
| Moscow        | Mon 29-May-2023 20:00 (08:00 PM) |
| Chennai       | Mon 29-May-2023 22:30 (10:30 PM) |
| Hangzhou      | Tue 30-May-2023 01:00 (01:00 AM) |
| Tokyo         | Tue 30-May-2023 02:00 (02:00 AM) |
| Sydney        | Tue 30-May-2023 03:00 (03:00 AM) |

Or in your local time:
* https://www.timeanddate.com/worldclock/fixedtime.html?msg=Node.js+Foundation+Performance%20Team+Meeting+2023-05-29&iso=20230529T1700
* or https://www.wolframalpha.com/input/?i=05PM+UTC%2C+May+29%2C+2023+in+local+time

## Links

* Minutes Google Doc: <https://docs.google.com/document/d/1YH4MB6l8QUgeRs1A0d_chOgmB9e_xwETmx9BQn18fZE/edit>

## Agenda

Extracted from **performance-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.

### nodejs/performance

* Performance of WHATWG ReadableStream.read() [#82](https://github.com/nodejs/performance/issues/82)
* Improve legacy CJS resolve [#73](https://github.com/nodejs/performance/issues/73)
* MakeCallback is very slow [#24](https://github.com/nodejs/performance/issues/24)
* fetch [#11](https://github.com/nodejs/performance/issues/11)

## Invited

* Performance team: @nodejs/performance

## Notes

The agenda comes from issues labelled with `performance-agenda` across **all of the repositories in the nodejs org**. Please label any additional issues that should be on the agenda before the meeting starts.

## Joining the meeting

* link for participants: https://zoom.us/j/94710602723

### PR DESCRIPTION
**Pull request by UlisesGascon**
close #85

### CODE REVIEW

File: meetings/2023-05-29.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Agenda
+
+## Announcements
+
+*Extracted from **performance-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.
+
+
+* https://github.com/nodejs/node/pull/48078
+  * It is providing a performance boost due the new Linux architecture
```
Comment (anonrig) [inline meetings/2023-05-29.md:24]:
```suggestion
* Libuv 1.45.0: https://github.com/nodejs/node/pull/48078
  * Thanks to @bnoordhuis and https://github.com/isc-projects which iiuc sponsored some of his work.
  * It is providing a performance boost due the new Linux architecture
```

File: meetings/2023-05-29.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+* https://github.com/nodejs/performance/issues/85#issuecomment-1563757752
+  * 30% boost at startup (v20.1.0 -> v20.2.0)
```
Comment (anonrig) [inline meetings/2023-05-29.md:27]:
```suggestion
* https://github.com/nodejs/performance/issues/85#issuecomment-1563757752
  * Thanks to @kvakil: 30% boost at startup (v20.1.0 -> v20.2.0) 
```

File: meetings/2023-05-29.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+*Extracted from **performance-agenda** labelled issues and pull requests from the **nodejs org** prior to the meeting.
+
+
+* https://github.com/nodejs/node/pull/48078
+  * It is providing a performance boost due the new Linux architecture
+
+* https://github.com/nodejs/performance/issues/85#issuecomment-1563757752
+  * 30% boost at startup (v20.1.0 -> v20.2.0) 
+
+* Ulises will help to host the next two meetings
```
Comment (anonrig) [inline meetings/2023-05-29.md:29]:
```suggestion
* Yagiz is leaving for paternity leave. Ulises will help to host the next two meetings.
```

File: meetings/2023-05-29.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  * We are a little bit slower in comparison with Bun and Deno
+  * It will be nice to see if Deno/Bun is using native implementation or JS.
+* Improve legacy CJS resolve #73
+  * The current approach is to use `simdjson` to perform the JSON parsing and increase the performance. See: https://github.com/nodejs/node/pull/47991
+  * @H4ad is working on this issue from now on
+* MakeCallback is very slow [#24](https://github.com/nodejs/performance/issues/24)
+  * We reached out to more people about it, including the TSC meeting.
+  * We need to do a deeper investigation
+  * This approach made by the `uWebSockets.js` is interesting. See: https://github.com/nodejs/performance/issues/24#issuecomment-1559993286
+  * Daniel & Yagiz will work soon in a new library to resolve the Websockets layer (compatible with Node.js)
```
Comment (anonrig) [inline meetings/2023-05-29.md:43]:
```suggestion
  * Daniel(@lemire) & Yagiz will work soon in a new library to resolve the Websockets layer (compatible with Node.js)
```

File: meetings/2023-05-29.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+* fetch [#11](https://github.com/nodejs/performance/issues/11)
+  * No news, we will remove it from the agenda
+  * Benchmarks and more info is needed before we can make any progress. 
+  * Contributions from the community are more than welcome here.
+
+## Q&A, Other
+
+*  Expensive `isTraceHTTPEnable` (https://github.com/nodejs/performance/issues/81) will be added back to the agenda.
```
Comment (anonrig) [inline meetings/2023-05-29.md:52]:
```suggestion
*  Expensive `isTraceHTTPEnable` (https://github.com/nodejs/performance/issues/81) is worth mentioning. @H4ad is working on it.
```

File: meetings/2023-05-29.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+* Vinícius will be nominated as Performance Team member
```
Comment (anonrig) [inline meetings/2023-05-29.md:53]: Can we move this item to the announcements?

```suggestion
* Vinícius is nominated/invited to the Performance Team. Reference: https://github.com/nodejs/performance/issues/86
```

**Review by anonrig — APPROVED** [review]

**Review by H4ad — APPROVED** [review]

**Pull request merged**